src/_utils.ts

```diff
@@ -106,7 +136,7 @@ export function setupOutgoing(outgoing, options, req, forward?) {
       requiresPort(outgoing.port, options[forward || "target"].protocol) &&
       !hasPort(outgoing.host)
         ? outgoing.host + ":" + outgoing.port
-        : outgoing.host;
+        : (outgoing.host as string | undefined);
```

Comment:
I used `as` here to keep the behavior as before, but I guess this should be:
```suggestion
        : outgoing.host ?? undefined;
```